examples/lighting-app/nrfconnect/main/Rpc.cpp

```diff
@@ -75,12 +85,13 @@ class Device final : public generated::Device<Device>
     {
         // TODO: Clear data from KVS
         DeviceLayer::ConfigurationMgr().InitiateFactoryReset();
+        // Reboot device after clearing data
+        k_timer_start(&reboot_timer, K_SECONDS(1), K_FOREVER);
         return pw::OkStatus();
     }
     pw::Status Reboot(ServerContext & ctx, const pw_protobuf_Empty & request, pw_protobuf_Empty & response)
     {
-        NVIC_SystemReset();
-        // WILL NOT RETURN
+        k_timer_start(&reboot_timer, K_SECONDS(1), K_FOREVER);
```

Comment:
a comment here as well of why we put this in a timer could be useful for future readers of the code who do not have the PR summary as a reference.

Reply:
I'll have a follow-up PR for the other platforms and will include this change.